test/e2e_test.go

```diff
@@ -468,6 +515,69 @@ func consumeMemory(client clientset.Interface, podName string) error {
 	return watchPodReadyStatus(client, metav1.NamespaceDefault, podName, currentPod.ResourceVersion)
 }
 
+func consumeWithInitContainerCPU(client clientset.Interface, podName string) error {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Spec: corev1.PodSpec{Containers: []corev1.Container{
+			{
+				Name:    podName,
+				Command: []string{"./consume-cpu/consume-cpu"},
+				Args:    []string{"--duration-sec=60", "--millicores=50"},
+				Image:   "gcr.io/kubernetes-e2e-test-images/resource-consumer:1.5",
+				Resources: corev1.ResourceRequirements{
+					Requests: map[corev1.ResourceName]resource.Quantity{
+						corev1.ResourceCPU: mustQuantity("100m"),
+					},
+				},
+			},
+		},
+			InitContainers: []corev1.Container{
+				{
+					Name:    "init-container",
+					Command: []string{"./consume-cpu/consume-cpu"},
```

Comment:
Please make it run for non-zero time so we can verify if we report metrics from init containers.